### DISCUSSION TITLE
Dynamic vs Static Formula

### DISCUSSION THREAD

**Discussion by mrnoisytiger** [Budibase/budibase#11661] [Help] [answered]
What is the difference between a dynamic and static formula? I see that I can use a static formula in some situations like a filter, but not a dynamic formula.

**Comment by melohagan** [comment, marked as answer]
Hey @mrnoisytiger 

I've updated the docs to include a description of the difference: https://docs.budibase.com/docs/formula#static-formulas

As you noted, static formulas can be filtered because the calculated value is actually saved into the database. 

Dynamic formulas on the other hand are only calculated when data is read. External databases only support dynamic formulas.

**Reply by mrnoisytiger** [reply]
Thanks for clarifying. I'm struggling to understand a little bit more on the interaction of a static formula still. 

I have a formula in my data table which is the difference between two columns A and B (C = A - B). Further, my app has a form which can edit `A` and `B`. The form has fields for `A` and `B` and the button action is to "Save Row" with the correct table set and a confirmation. 

I've tried multiple times updating `A` and `B` using the form and have confirmed that the changes are taking place in the data. However, `C` is not being calculated. Since I'm saving the row, I'm expecting that it does get calculated upon save. Any idea of why I'm experiencing this? 

For information, I can make the static formula recalculate by changing the expression (or switching to dynamic and back to static), but that seems to be it.

**Reply by melohagan** [reply]
I did notice there was a bug relating to switching between dynamic and static formulas: https://github.com/Budibase/budibase/issues/11669

You could try deleting the formula field, and recreating making sure to save it as static initially
